### PR TITLE
Release/4.0.0-rc.2

### PR DESCRIPTION
**Pull request by aboedo**
- Second RC for RevenueCat framework v4 🎉
    100% Swift framework + ObjC support.

[Full Changelog](https://github.com/revenuecat/purchases-ios/compare/4.0.0-beta.rc.1...4.0.0-rc.2)
- See our [RevenueCat V4 API update doc](docs/V4_API_Updates.md) for API updates.

RC 2 introduces the following updates:

### API changes:

- Removed `SubscriptionPeriod.Unit.unknown`. Subscriptions with empty `SubscriptionPeriod` values will have `nil` `subscriptionPeriod` instead.
- Removed `StoreProductDiscount.none`, since it wasn't needed.

### Documentation: 

- Documentation is now using DocC and it can be found in https://revenuecat-docs.netlify.app/documentation/Revenuecat. 
- We've made several improvements to docstrings and added a few landing pages for the most important sections of the SDK. 

### Migration fixes

- Fixed a few instances where Xcode's automatic migration tools wouldn't correctly update the code.

### Other changes: 
- There are lots of under the hood improvements. If you see any issues we'd appreciate [bug reports](https://github.com/RevenueCat/purchases-ios/issues/new?assignees=&labels=bug&template=bug_report.md&title=)!

### Changes from previous RC

These changes add to all of the changes from beta RC 1, [listed here.](https://github.com/RevenueCat/purchases-ios/blob/4.0.0-rc.1/CHANGELOG.latest.md).

### Requirements: 
- https://github.com/RevenueCat/purchases-ios/pull/1213

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -81,7 +81,7 @@ platform :ios do
     
     files_to_update = [
       '../RevenueCat.podspec',
-      '../Purchases/Public/Purchases.swift',
+      '../Purchases/Purchasing/Purchases.swift',
```
Comment (aboedo) [inline fastlane/Fastfile]: fixed this path, it was outdated after I moved stuff out of `Public`

File: CHANGELOG.latest.md [hunk 2]
```diff
@@ -1,33 +1,28 @@
...
+### Migration fixes
+
+- Fixed a few instances where Xcode's automatic migration tools wouldn't correctly update the code.
 
 ### Other changes: 
 - There are lots of under the hood improvements. If you see any issues we'd appreciate [bug reports](https://github.com/RevenueCat/purchases-ios/issues/new?assignees=&labels=bug&template=bug_report.md&title=)!
+
+### Changes from previous RC
+
+These changes add to all of the changes from beta RC 1, [listed here.](https://github.com/RevenueCat/purchases-ios/blob/4.0.0-rc.1/CHANGELOG.latest.md).
```
Comment (aboedo) [inline CHANGELOG.latest.md:28]: all auto-generated from this point on, other than the podspec and fastfile at the very bottom of the PR

**Comment by taquitos** [comment]
I'm having a rough time using the UI in GitHub to view this. Unfortunately, I think we should separate these into two PRs in the future. One for docs, then one for code.

**Comment by aboedo** [comment]
@taquitos I went with this approach because it was very simple to set up, but I do see the inconvenience with reviews. 
Note that the doc generating step is automated, done by `bump`, exactly the same as we auto-update the version number and the `RevenueCat-Swift.h` file. 
The only manual step in this flow is updating the changelog... Which I think we should start doing in the PRs themselves, not in a separate release PR. It takes a lot of work to compile a changelog and make sure that everything makes it in, but if we're doing it in every PR (we could have a check for that), it'd be trivial. 

I didn't add it to the auto-deploy step instead for the same reason that we have the version number bumps and the `RevenueCat-Swift.h` file done in the `bump` step: we want to deploy from a tag, and that currently means creating the tag locally first and then pushing. 

An alternative approach would be to have a `create_tag_and_deploy` lane, which runs: 
- version bump
- `RevenueCat-Swift.h` generation
- docs compilation
Then auto-commits without review, then creates a tag and deploy. I guess the "auto-commit without review" part is the part that makes me a bit nervous, and the reason we're not doing this entirely automated right now. 

Thoughts?

**Review by taquitos — APPROVED** [review]
🚢

**Comment by taquitos** [comment]
> @taquitos I went with this approach because it was very simple to set up, but I do see the inconvenience with reviews. Note that the doc generating step is automated, done by `bump`, exactly the same as we auto-update the version number and the `RevenueCat-Swift.h` file. The only manual step in this flow is updating the changelog... Which I think we should start doing in the PRs themselves, not in a separate release PR. It takes a lot of work to compile a changelog and make sure that everything makes it in, but if we're doing it in every PR (we could have a check for that), it'd be trivial.
> 
> I didn't add it to the auto-deploy step instead for the same reason that we have the version number bumps and the `RevenueCat-Swift.h` file done in the `bump` step: we want to deploy from a tag, and that currently means creating the tag locally first and then pushing.
> 
> An alternative approach would be to have a `create_tag_and_deploy` lane, which runs:
> 
> * version bump
> * `RevenueCat-Swift.h` generation
> * docs compilation
>   Then auto-commits without review, then creates a tag and deploy. I guess the "auto-commit without review" part is the part that makes me a bit nervous, and the reason we're not doing this entirely automated right now.
> 
> Thoughts?

High-level, I think it makes sense to let CI handle version bump, .h generation, and docs. Anything that is autogenerated should be something the CI handles. That doesn't mean we have to let it auto-approve, though. I think having CI generate a release PR that we approve might be the way to go. Once approved, the CI can take the next steps of merging and tagging.

This definitely will require more thought to get it right, though. In the meantime, I'm totally fine with what we have.

**Pull request merged**